src/java_bytecode/expr2java.h

```diff
@@ -50,4 +53,52 @@ class expr2javat:public expr2ct
 std::string expr2java(const exprt &expr, const namespacet &ns);
 std::string type2java(const typet &type, const namespacet &ns);
 
+/// Convert floating point number to a string without printing
+/// unnecessary zeros. Prints decimal if precision is not lost.
+/// Prints hexadecimal otherwise, and/or java-friendly NaN and Infinity
+template <typename float_type>
+std::string floating_point_to_java_string(float_type value)
+{
+  const auto is_float = std::is_same<float_type, float>::value;
+  static const std::string class_name = is_float ? "Float" : "Double";
+  if(std::isnan(value))
+    return class_name + ".NaN";
+  if(std::isinf(value) && value >= 0.)
+    return class_name + ".POSITIVE_INFINITY";
+  if(std::isinf(value) && value <= 0.)
+    return class_name + ".NEGATIVE_INFINITY";
+  // Using ostringstream to get string without trailing zeros
+  const std::string decimal = [&] { // NOLINT
+    std::ostringstream raw_stream;
+    raw_stream << value;
+    const auto raw_decimal = raw_stream.str();
+    if(raw_decimal.find('.') == std::string::npos)
+      return raw_decimal + ".0";
+    return raw_decimal;
+  }();
```

Comment:
I think a new line after each lambda call would make this function a lot more readable

Reply:
I think not. These kinds of lambdas serve the same purpose as new lines. The benefit of using them instead is giving smaller scopes to variable names and enabling more `const`. They are already indented, so I don't think additional new lines would help.

Reply:
The rest of the code base doesn't use lambdas like this and we don't need more inconsistency.  Also if I have to choose between a slightly smaller scope on variables and simpler code which works with more debuggers and profiling tools; the choice is obvious.  Please don't PR code like this.